### PR TITLE
Add a new `nix store add` command

### PR DESCRIPTION
**Pull request by thufschmitt**
# Motivation

Deprecate `nix store add-file` and `nix store add-path`, and replace them with a single `nix store add` command.

# Context

Fix https://github.com/NixOS/nix/issues/8915

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: src/nix/add-to-store.cc [hunk 1]
```diff
@@ -23,6 +34,23 @@ struct CmdAddToStore : MixDryRun, StoreCommand
...
+        addFlag({
+            .longName  = "mode",
+            .shortName = 'n',
+            .description = R"(
+    How to compute the hash of the input.
+    One of:
+
+    - `nar` (the default): Serialises the input as an archive (following the [_Nix Archive Format_](https://edolstra.github.io/pubs/phd-thesis.pdf#page=101)) and passes that to the hash function.
+
+    - `flat`: Assumes that the input is directly a single file and directly passes it to the hash function;
```
Comment (fricklerhandwerk) [inline src/nix/add-to-store.cc:47]:
```suggestion
    - `flat`: Assumes that the input is a single file and directly passes it to the hash function;
```

**Review by Ericson2314 — APPROVED** [review]
Looks good! Just needs change log entry I think.

I forgot whether we talked about it. I think this should support text hashing too, but it is easiest to do that after https://github.com/NixOS/nix/pull/9325 so I am OK not doing it for now.

**Comment by thufschmitt** [comment]
> Just needs change log entry I think.

Ah good catch, I always forget about these.

@fricklerhandwerk, I received a notification with a suggestion from you which disappeared from here. It was good so I've applied it nonetheless.

**Comment by Ericson2314** [comment]
Oh sorry one more thing I missed: `add-path.md` should become `add.md` I think?

**Comment by thufschmitt** [comment]
Grmbl, indeed :)

**Pull request merged**